### PR TITLE
Idea upvote model - many to many pivot table

### PR DESCRIPTION
**Pull request by MikiWolfe**
Added the Idea Upvote Model that’s a many to many pivot table between User and Idea,

### CODE REVIEW

File: models/index.js [hunk 1]
```diff
@@ -32,6 +33,23 @@ Idea.hasMany(Comment, {
...
+  as: 'upvote'
+});
+
+User.belongsTo(Idea, {
+  foreignKey: 'user_id',
+  through : IdeaUpvote,
+  as: 'upvotes'
+});
+
+Idea.belongsTo(User, {
```
Comment (jmichaelbrown8) [inline models/index.js:61]: This might not be needed.

Comment (ac524) [reply models/index.js]: @MikiWolfe These three relations should instead be replace with 2 `belongsToMany` relations. See my examples for connecting `User` to `Space` through the `SpaceMember` model. You'll need a similar set up to connect `Idea` to `User` through the `IdeaUpvote` model.

I think we will also need direct relations between `Idea` and the `IdeaUpvote` table using the standard `Idea hasMany IdeaUpvote` (no through)

**Review by ac524 — CHANGES_REQUESTED** [review]

File: models/index.js [hunk 2]
```diff
@@ -58,6 +59,23 @@ Idea.hasMany(Comment, {
...
+  foreignKey: 'user_id',
+  through : IdeaUpvote,
+  as: 'upvoter'
+});
+
+Idea.hasMany(IdeaUpvote, {
+  foreignKey: 'idea_id',
+});
+
+User.belongsTo(Idea, {
```
Comment (ac524) [inline models/index.js:72]: This should be `User.belongsToMany`

File: models/IdeaUpvote.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const {Model, DataTypes} = require('sequelize');
+const sequelize = require('../config/connection');
+
+class IdeaUpvote extends Model {}
+
+IdeaUpvote.init(
+  {
+    id: {
```
Comment (ac524) [inline models/IdeaUpvote.js:8]: We'll want to remove this `id` field, The relations in the `index.js` will configure the needed foreign key columns for us

**Review by ac524 — CHANGES_REQUESTED** [review]

**Review by jmichaelbrown8 — APPROVED** [review]
Looks good to me!

**Review by ac524 — APPROVED** [review]
🙌  Great success!

**Pull request merged**